pkg/controller/networkaddonsconfig/networkaddonsconfig_controller_test.go

```diff
@@ -0,0 +1,166 @@
+package networkaddonsconfig
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/names"
+	"github.com/kubevirt/cluster-network-addons-operator/pkg/render"
+)
+
+type checkUnit struct {
+	key           string
+	shouldExist   bool
+	expectedValue string
+}
+
+var _ = Describe("Networkaddonsconfig", func() {
+	const testDataFile = "testdata/data.yaml"
+	const numOfObjectsInDataFile = 4
+
+	Context("When CR is not labeled", func() {
+		var objs []*unstructured.Unstructured
+
+		BeforeEach(func() {
+			var err error
+			renderData := render.MakeRenderData()
+			objs, err = render.RenderTemplate(testDataFile, &renderData)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(objs)).To(Equal(numOfObjectsInDataFile))
+
+			crLabels := map[string]string{}
+			err = updateObjectsLabels(crLabels, objs)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("Should find only default relation labels", func() {
+			appLabelKeys := []checkUnit{
+				{
+					key:           names.COMPONENT_LABEL_KEY,
+					shouldExist:   true,
+					expectedValue: names.COMPONENT_LABEL_DEFAULT_VALUE,
+				},
+				{
+					key:           names.MANAGED_BY_LABEL_KEY,
+					shouldExist:   true,
+					expectedValue: names.MANAGED_BY_LABEL_DEFAULT_VALUE,
+				},
+				{
+					key:           names.PART_OF_LABEL_KEY,
+					shouldExist:   false,
+					expectedValue: "Invalid",
+				},
+				{
+					key:           names.VERSION_LABEL_KEY,
+					shouldExist:   false,
+					expectedValue: "Invalid",
+				},
+			}
+
+			checkObjectsRelationshipLabels(objs, appLabelKeys)
+		})
+	})
+	Context("When CR is labeled", func() {
+		var objs []*unstructured.Unstructured
+		var crLabels map[string]string
+		const expectedComponentLabel = "component_unit_tests"
+		const expectedManagedByLabel = "managed_by_unit_tests"
+		const expectedPartOfLabel = "part_of_unit_tests"
+		const expectedVersionLabel = "version_of_unit_tests"
+
+		BeforeEach(func() {
+			var err error
+			renderData := render.MakeRenderData()
+			objs, err = render.RenderTemplate(testDataFile, &renderData)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(objs)).To(Equal(numOfObjectsInDataFile))
```

Comment:
```suggestion
			Expect(objs).To(HaveLen(numOfObjectsInDataFile))
```
also, why do we need this? if RenderTemplate fails to render some objects, won't it return an error?

Reply:
yep it will, it just so it will be predictable when we add stuff to the template
i can remove it if needed, it was added just an extra validation

Reply:
I don't think it's needed, provided that `RenderTemplate` fails if not all objects are rendered. 
I don't want to update this param every time I add an object to the data.yaml if it's not necessary